projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts

```diff
@@ -283,29 +315,48 @@ export class IgxButtonGroupComponent implements AfterViewInit {
      * ```
      */
     public deselectButton(index: number) {
-        if (this.buttons.toArray()[index]._el.nativeElement.getAttribute('data-togglable') === 'false'
-            || this.buttons.toArray()[index]._el.nativeElement.classList.contains('igx-button--disabled')) {
+        const button = this.buttons[index];
+        const buttonElement = button.nativeElement;
+
+        if (buttonElement.classList.contains('igx-button--disabled')) {
             return;
         }
-        const buttonElement = this.buttons.toArray()[index]._el.nativeElement;
+
         this.selectedIndexes.splice(this.selectedIndexes.indexOf(index), 1);
-        buttonElement.setAttribute('data-selected', false);
-        this.onUnselect.emit({ button: this.buttons.toArray()[index], index });
-        this.values[index].selected = false;
+
+        this._renderer.setAttribute(buttonElement, 'data-selected', 'false');
+        this._renderer.setAttribute(buttonElement, 'aria-pressed', 'false');
+        this._renderer.removeClass(buttonElement, 'igx-button-group__item--selected');
+
+        this.onUnselect.emit({ button: button, index: index });
+
+        const indexInViewButtons = this.viewButtons.toArray().indexOf(button);
+        if (indexInViewButtons !== -1) {
+            this.values[indexInViewButtons].selected = false;
+        }
     }
 
     /**
      * @hidden
      */
     public ngAfterViewInit() {
-        // initial selection
-        setTimeout(() => {
-            this.buttons.forEach((button, index) => {
-                if (!button.disabled && button._el.nativeElement.getAttribute('data-selected') === 'true') {
-                    this.selectButton(index);
-                }
-            });
-        }, 0);
+        // initial configuration
+        this.buttons.forEach((button, index) => {
```

Comment:
What happens when we have arbitrary number of buttons rendered? Consider the case when somebody templates the buttons with `ngFor` and the input array is being changed "runtime". I think none of the classes and handlers will be applied for new elements as this is initialize once and forget. This should be moved inside the [`changes`](https://angular.io/api/core/QueryList#changes) emitter for both query lists and appropriate updates should be applied when a button is added/removed from the template.